### PR TITLE
Refactor: Move LDAP auth config frontend registration to OSS

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
We moved the LDAP authentication debug page to /admin/authentication. The /admin/authentication route was behind saml feature, which is not the case for running in OSS. This moves LDAP auth reg to the OSS repo and removes the registration in enterprise.

Fixes #
https://github.com/grafana/grafana/issues/73884

**Special notes for your reviewer:**
this assumes we do not have any other need for registering the LDAP component. (it is checked by ldap/enabled)

Enterprise PR - https://github.com/grafana/grafana-enterprise/pull/5692

### CODE REVIEW

File: pkg/services/navtree/navtreeimpl/admin.go [hunk 1]
```diff
@@ -15,7 +15,6 @@ func (s *ServiceImpl) getAdminNode(c *contextmodel.ReqContext) (*navtree.NavLink
 	hasAccess := ac.HasAccess(s.accessControl, c)
 	hasGlobalAccess := ac.HasGlobalAccess(s.accessControl, s.accesscontrolService, c)
 	orgsAccessEvaluator := ac.EvalPermission(ac.ActionOrgsRead)
-	authConfigUIAvailable := s.license.FeatureEnabled("saml")
```
Comment (eleijonmarck) [inline pkg/services/navtree/navtreeimpl/admin.go:18]: not needed as a check for registration the route

File: public/app/features/auth-config/types.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ export interface AuthProviderInfo {
   protocol: string;
   displayName: string;
   configPath?: string;
+  needEnterpriseLicense?: boolean;
```
Comment (eleijonmarck) [inline public/app/features/auth-config/types.ts:9]:
```suggestion
```

File: public/app/routes/routes.tsx [hunk 3]
```diff
@@ -299,19 +299,15 @@ export function getAppRoutes(): RouteDescriptor[] {
...
-        ? SafeDynamicImport(
-            () => import(/* webpackChunkName: "AdminAuthentication" */ 'app/features/auth-config/AuthConfigPage')
-          )
-        : () => <Redirect to="/admin" />,
-    },
     {
       path: '/admin',
       component: () => <NavLandingPage navId="cfg" header={<ConnectionsRedirectNotice />} />,
     },
+    {
```
Comment (eleijonmarck) [inline public/app/routes/routes.tsx:306]: moved it after the /admin route registration.

what is changed is this line:
```go
component: config.licenseInfo.enabledFeatures?.saml
        ? SafeDynamicImport(
            () => import(/* webpackChunkName: "AdminAuthentication" */ 'app/features/auth-config/AuthConfigPage')
          )
        : () => <Redirect to="/admin" />,
```        
to:
```go
component: SafeDynamicImport(() => import(/* webpackChunkName: "AdminAuthentication" */ 'app/features/auth-config/AuthConfigPage')),
```

**Review by eleijonmarck — COMMENTED** [review]
main gist of this change: https://github.com/grafana/grafana/pull/73941/files#r1307510973

File: public/app/routes/routes.tsx [hunk 4]
```diff
@@ -299,19 +299,15 @@ export function getAppRoutes(): RouteDescriptor[] {
...
+      path: '/admin/authentication',
+      roles: () => contextSrv.evaluatePermission(() => ['Admin', 'ServerAdmin'], [AccessControlAction.ServiceAccountsWrite]),
```
Comment (kalleep) [inline public/app/routes/routes.tsx:308]: Unintentional change of action?

Comment (eleijonmarck) [reply public/app/routes/routes.tsx:308]: yeas! my bad 🙏

File: public/app/features/auth-config/index.ts [hunk 5]
```diff
@@ -21,13 +23,13 @@ export function getRegisteredAuthProviders(): AuthProviderInfo[] {
 }
 
 export function getAuthProviderInfo(provider: string) {
-  return registeredAuthProviders.find((p) => p.id === provider);
+  return registeredAuthProviders.find((p: AuthProviderInfo) => p.id === provider);
```
Comment (alexanderzobnin) [inline public/app/features/auth-config/index.ts:26]: I don't think type needs to be explicitly mentioned here, it's correctly evaluated as `AuthProviderInfo` implicitly from `registeredAuthProviders`

File: public/app/routes/routes.tsx (hunk 3 above)
Comment (alexanderzobnin) [reply public/app/routes/routes.tsx:306]: Could you double check what's a behaviour when LDAP is not enabled and edition is OSS (no SAML)? Is it empty page? I guess no CTA should be shown there.

**Review by alexanderzobnin — COMMENTED** [review]
Looks good, we just need to make sure page works correctly if there's no SAML available and LDAP isn't enabled.

**Review by alexanderzobnin — APPROVED** [review]

**Comment by kenntun1** [comment]
@eleijonmarck eleijonmarck 
    [auth.ldap]
    enabled = true
    config_file = /etc/grafana/ldap.toml
    allow_sign_up = true

I can login LDAP users but said screen are not visible on Grafana OSS 10.1.0 or 10.1.1

**Comment by eleijonmarck** [comment]
@kenntun1 we expect it to be part of the 10.2

**Pull request merged**